### DISCUSSION TITLE
boot into black screen after installation on manjaro linux

### DISCUSSION THREAD

**Discussion by HilariousJon** [prasanthrangan/hyprdots#1770] [General]
# Description

I opened a post on manjaro official forum and describe my situation there. Really hope that someone can help me. 

the post can be refer via this [link](https://forum.manjaro.org/t/system-boot-into-black-screen-after-installing-hyprland/167131/15)

**Comment by kRHYME7** [comment]
Can you run `Hyprland ` in the tty?

Also sddm did not display after boot?

**Comment by HilariousJon** [comment]
Dear @kRHYME7 
oh yes, I can run `Hyprland` in tty, it makes things appear, but how to make it appear automatically, so I don't need tty? sddm runs well as I typed `systemctl status display-manager` it shows that it works well. And does it mean that my kde crash? How can I fix that?

**Comment by HilariousJon** [comment]
and I notice that I run `code` in terminal, things will crash, how to solve it?

**Comment by kRHYME7** [comment]
systemctl enable --now sddm

**Comment by HilariousJon** [comment]
@kRHYME7 
`systemctl enable --now sddm` didn't work. and I am really curious about why `vscode` will crash in hyprland. the error message of that is:
```
Warning: 'ozone-platform-hint' is not in the list of known options, but still passed to Electron/Chromium
```

**Comment by kRHYME7** [comment]
does sddm show after boot?

**Comment by HilariousJon** [comment]
@kRHYME7 
https://vimeo.com/manage/videos/1001468638 
this video can show my current situation. And I just find that I cannot open my **obsidian** either. with error logs to be:

```
❯ obsidian
2024-08-22 05:39:42 Loading main app package /usr/lib/obsidian/obsidian.asar
[22752:0822/133942.594465:ERROR:object_proxy.cc(576)] Failed to call method: org.freedesktop.DBus.Properties.Get: object_path= /org/freedesktop/portal/desktop: org.freedesktop.DBus.Error.InvalidArgs: No such interface “org.freedesktop.portal.FileChooser”
[22752:0822/133942.594883:ERROR:select_file_dialog_linux_portal.cc(290)] Failed to read portal version property
2024-08-22 05:39:42 Checking for update using Github
2024-08-22 05:39:42 Failed to check for update using Github (net::ERR_CONNECTION_REFUSED)
Error: net::ERR_CONNECTION_REFUSED
    at SimpleURLLoaderWrapper.<anonymous> (node:electron/js2c/browser_init:2:114482)
    at SimpleURLLoaderWrapper.emit (node:events:519:28)
2024-08-22 05:39:42 Checking for update using obsidian.md
[22752:0822/133943.428441:ERROR:gpu_process_host.cc(997)] GPU process exited unexpectedly: exit_code=139
2024-08-22 05:39:43 Failed to check for update using obsidian.md (Unexpected end of JSON input)
SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at runUpdate (/usr/lib/obsidian/app.asar/main.js:183:21)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async update (/usr/lib/obsidian/app.asar/main.js:149:3)
[22752:0822/133944.317927:ERROR:gpu_process_host.cc(997)] GPU process exited unexpectedly: exit_code=139
[22752:0822/133945.133258:ERROR:gpu_process_host.cc(997)] GPU process exited unexpectedly: exit_code=139
[22752:0822/133946.172545:ERROR:gpu_process_host.cc(997)] GPU process exited unexpectedly: exit_code=139
[22752:0822/133947.205816:ERROR:gpu_process_host.cc(997)] GPU process exited unexpectedly: exit_code=139
[22752:0822/133948.260614:ERROR:gpu_process_host.cc(997)] GPU process exited unexpectedly: exit_code=139
[22752:0822/133948.719800:ERROR:gpu_process_host.cc(997)] GPU process exited unexpectedly: exit_code=139
[22752:0822/133949.239875:ERROR:gpu_process_host.cc(997)] GPU process exited unexpectedly: exit_code=139
[22752:0822/133949.691930:ERROR:gpu_process_host.cc(997)] GPU process exited unexpectedly: exit_code=139
[22752:0822/133949.691982:FATAL:gpu_data_manager_impl_private.cc(448)] GPU process isn't usable. Goodbye.
[1]    22752 trace trap (core dumped)  obsidian
```

--- 

As you can see in the video, it attains the asus logo appearing and stuck at that position. I am not sure if sddm shows.

--- 

Another update, if I directly type the password blindly under the asus logo, it will login. so it means that the login page is totally not shown.

**Comment by kRHYME7** [comment]
You got different  issues but maybe can be streamlined that you're using an asus laptop. 

problems:
1. sddm did not show up
2. electron apps like 
2.a vscode won't  open
2.b obisidian (according to the logs you got errors rendering them, I'm not sure but maybe something to do with gpu)

I guess you should follow  this first, specially using an Asus laptop https://asus-linux.org/guides/arch-guide/

thankfully we got a script to do the things inside that tutorial, just run `  Hyde inject asus_patch  `  

Take note that I don't know if this could fix your issue, but I'm pretty sure these fix/patch are prerequisites for your arch installation, so it should be good to have it. 


Also, some apps eg vscode got flags to disable gpu rendering you can test it byr running `--disable-gpu ` .

**Comment by HilariousJon** [comment]
@kRHYME7 
Yes, I checked the crash list of the system, and it seems that electron directly crashed in my hyprland desktop env.

and actually I found 

> To ease the installation you can use other distro based on Arch, only Manjaro is highly not recommend (it is not really based on Arch and might not be compatible with points above and things might break with upgrades).

:joy: seems not quite a problem caused by asus I guess? I am not using rog, other kind of asus computer. I found another issue in this repo, whose screenshots shows that it install hyprland on manjaro pretty well.

refer it here: https://github.com/prasanthrangan/hyprdots/issues/558

any other clues that may cause such problems?

--- 

`--disable-gpu` didn't help. I think it maybe relate to some global configuration problem. I also notice that the cutie pictures and text in the terminal doesn't show normally, I don't know whether it is you guys updates that make a difference than before. You can see it in the picture below.

![IMG_20240822_163726](https://github.com/user-attachments/assets/5cdca524-7fb0-4bee-b84b-8e1b39fc892a)

**Comment by kRHYME7** [comment]
> 
![image](https://github.com/user-attachments/assets/01ca209d-ae21-41dd-8d55-eba730354090)

Looks normal to me and nothing is change here, just the ` pokemon-colorscripts ` command

>  I think it maybe relate to some global configuration problem

this repo generally sets the Sessions envs (because user logs in > hyprland opens with the env vars) 

I guess you should try to fix your sddm errrors first, how about try to install/change a theme using ` Hyde sddm install `
![image](https://github.com/user-attachments/assets/85316818-0e6b-479c-bc7f-bcd6cde35702)

about other rendering stuff are you using nvidia? too ? You check out nvidia page of the hyprland wiki

**Comment by HilariousJon** [comment]
@kRHYME7 
I tried, running `Hyde sddm install`, nothing change. Should have changed if it is caused by missing sddm theme.

I have a nvdia card, I ll go and check hyprland wiki.

---

about the pokemon script, I just don't know how to operate on it :joy: any wiki?

**Comment by thyeun** [comment]
@HilariousJon To open your ```code```, go into your ```~/.config/``` and create a file ```code-flags.conf``` and paste in the code below.

```
--ozone-platform-hint=wayland
--use-gl=egl
```

**Comment by thyeun** [comment]
@HilariousJon To solve your ```sddm```, re-install it with ```yay -S sddm-git``` (during install, it will ask you to remove ```sddm```, due to conflict, you just ```Yes``` to remove).

**Comment by HilariousJon** [comment]
@thyeun 
have tried reinstall `sddm`, not working. and the solution for `code` didn't work as well.  The error message for `code` become:
```
❯ code
Warning: 'ozone-platform-hint' is not in the list of known options, but still passed to Electron/Chromium.
Warning: 'use-gl' is not in the list of known options, but still passed to Electron/Chromium.
```
and as a note, I guess all electron related stuff cannot load. I checked the list of **Crashed Process Viewer** and it shows that both `code` and `electron` will crash.

**Comment by thyeun** [comment]
@HilariousJon Your ```code``` is vs code or ```OSS-code```?

**Comment by HilariousJon** [comment]
@thyeun 

vs code

**Comment by thyeun** [comment]
@HilariousJon If vs code it should be workable. As you can see my vs code below. Brave also workable.

![240822_18h20m28s_screenshot](https://github.com/user-attachments/assets/2bc28d37-6b70-4559-8721-879c24ed90e0)

**Comment by HilariousJon** [comment]
@thyeun 
no idea :joy:, its just super weird. the log shows that it indeed crash. Check the error message above. I also can see that when I press `linux+A` I can see clearly that `visual studio code` is in my software list. But I cannot open it either by running terminal command nor click on the icon.

--- 

Worth noting that, sublime text and kate are workable. In my whole computer, only obsidian and vscode are not able to load. (Maybe thats because I have few software installed before in my manjaro kde). Oh, maybe some kde stuff are not able to load.

**Comment by thyeun** [comment]
@HilariousJon You re-install ```sddm``` or re-install with ```sddm-git```? if you just reinstall ```sddm```, your ```sddm``` still wont work, you need to install ```sddm-git```

**Comment by HilariousJon** [comment]
@thyeun 
I re-install `sddm-git`. But let me try again, as I can't really remember some of the error message.

**Comment by thyeun** [comment]
> @thyeun no idea 😂, its just super weird. the log shows that it indeed crash. Check the error message above. I also can see that when I press `linux+A` I can see clearly that `visual studio code` is in my software list. But I cannot open it either by running terminal command nor click on the icon.

Do you do exactly what i mentioned by created the file ```code-flags.conf```?

**Comment by thyeun** [comment]
> @thyeun I re-install `sddm-git`. But let me try again, as I can't really remember some of the error message.

Remember to reboot after install, not just login/logout

**Comment by HilariousJon** [comment]
> I re-install sddm-git. But let me try again, as I can't really remember some of the error message.

Yes I have rebooted.

**Comment by HilariousJon** [comment]
> Do you do exactly what i mentioned by created the file code-flags.conf?

yes, I have. the following are current content:

```
# --ozone-platform-hint=wayland
# --disable-gpu
--ozone-platform-hint=wayland
--use-gl=egl
```

**Comment by thyeun** [comment]
@HilariousJon ```sudo pacman -Qs nvidia```, what is your output? if you install correctly as below, everything should work out of the box

![240822_18h39m30s_screenshot](https://github.com/user-attachments/assets/a82618b3-45b2-4743-aa5c-552396792b8b)

**Comment by HilariousJon** [comment]
here are my outputs for `sudo pacman -Qs nvidia`, I guess no particular error?

```
❯ sudo pacman -Qs nvidia
local/egl-wayland 4:1.1.13.1-1
    EGLStream-based Wayland external platform
local/ffnvcodec-headers 12.2.72.0-1
    FFmpeg version of headers required to interface with Nvidias codec APIs
local/lib32-libvdpau 1.5-2
    Nvidia VDPAU library
local/lib32-nvidia-utils 550.107.02-1
    NVIDIA drivers utilities (32-bit)
local/lib32-vulkan-nouveau 1:24.1.6-1
    Open-source Vulkan driver for Nvidia GPUs - 32-bit
local/libvdpau 1.5-3
    Nvidia VDPAU library
local/libxnvctrl 550.107.02-1
    NVIDIA NV-CONTROL X extension
local/mhwd-nvidia 550.107.02-1
    MHWD module-ids for nvidia 550.107.02
local/mhwd-nvidia-390xx 390.157-14
    MHWD module-ids for nvidia 390.157
local/mhwd-nvidia-470xx 470.256.02-4
    MHWD module-ids for nvidia 470.256.02
local/nvidia-dkms 550.107.02-1
    NVIDIA drivers - module sources
local/nvidia-settings 550.107.02-1
    Tool for configuring the NVIDIA graphics driver
local/nvidia-utils 550.107.02-1
    NVIDIA drivers utilities
local/vulkan-nouveau 1:24.1.6-1
    Open-source Vulkan driver for Nvidia GPUs
local/xf86-video-nouveau 1.0.17-3 (xorg-drivers)
    Open Source 3D acceleration driver for nVidia cards
```

---

hmm, seems mine mix a lot with open source one.

**Comment by thyeun** [comment]
@HilariousJon Since you have mix nvidia driver, change the below to and remove ```--use-gl=egl```.

```
--ozone-platform-hint=default
```

**Comment by HilariousJon** [comment]
@thyeun 

after the change, I have: 

```
❯ source ~/.config/code-flags.conf
zsh: command not found: --ozone-platform-hint=default
❯ code
Warning: 'ozone-platform-hint' is not in the list of known options, but still passed to Electron/Chromium.
```

**Comment by thyeun** [comment]
@HilariousJon ```code``` not open? if previous with ```wayland```, ```code``` is open, than it mean workable, the message is just a warning.

**Comment by HilariousJon** [comment]
@thyeun 

it hasn't. haven't opened so far.

---

actually I have tried several flags before, one of them indeed opened a blank window, but quickly terminated. I try to find it now.

---

should be this one: 

```
code --ozone-platform=x11

```

**Comment by thyeun** [comment]
@HilariousJon My advised is, dont using this dotfile, since your system abit messy, we dont know which and which driver are useful to your current system. Best use for this dotfile is fresh installation.

**Comment by HilariousJon** [comment]
@thyeun 

Could you be more specific? What dotfile? Maybe I can do a timeshift and try to deal with those drivers.

**Comment by thyeun** [comment]
Hyprdots is a dotfile

**Comment by HilariousJon** [comment]
@thyeun @kRHYME7 
but i guess currently it is hard to revert it to original kde :sob:.

**Comment by HilariousJon** [comment]
It seems that all electron related apps are not able to run. It should be a general problem.

**Comment by kRHYME7** [comment]
>  but i guess currently it is hard to revert it to original kde 😭.


you still can and can override the dotfiles ( configuration files ) from Hyprdots repo. you just need to find a way to start your plasma desktop. you can try to follow this  https://wiki.archlinux.org/title/SDDM if you want to fix sddm. 


about your electron issues, you can check if it's the hyprdots environment or not if you can open your plasma-desktop and see if electron apps runs.

**Comment by thyeun** [comment]
> It seems that all electron related apps are not able to run. It should be a general problem.

For sure it is only happen to you. And it is not general problem, all my electron apps working according what i want, and it done well in my system (nvidia + wayland + hyrpland + Hyprdots).

Why you feel hard to revert back? since you have timeshift.

**Comment by HilariousJon** [comment]
thats the problem :joy: I didn't save a timeshift..... I wasn't that familiar with the system, since what I took kde as a tool of startiing vscode :joy:

--- 

> For sure it is only happen to you. And it is not general problem, all my electron apps working according what i want, and it done well in my system (nvidia + wayland + hyrpland + Hyprdots).

I found something on electron repo:

https://github.com/electron/electron/issues/39449

**Comment by thyeun** [comment]
@HilariousJon Few things need you to be verify.

1. Your KDE is wayland or x11?
2. Previous SDDM in KDE working fine?

I might possible guide you to get back your KDE, if you have both info above

**Comment by HilariousJon** [comment]
Hi @thyeun 

1. kde is x11
2. previous sddm works fine within kde

**Comment by thyeun** [comment]
@HilariousJon Good info, so check below.

1. Go ```/etc/sddm.conf.d/```, check how many files inside?

Best if you can show a screenshot.

**Comment by HilariousJon** [comment]
```
❯ ls -al /etc/sddm.conf.d/
.rw-r--r--  23 root  2 5月  05:09  00_manjaro_settings.conf
.rw-r--r-- 199 root 22 8月  20:06  kde_settings.conf
.rw-r--r-- 199 root 22 8月  20:06  kde_settings.t2.bkp
.rw-r--r--  31 root  2 5月  05:09  virtualkeyboard.conf
```

**Comment by thyeun** [comment]
@HilariousJon Keep ```00_manjaro_settings.conf``` and ```virtualkeyboard.conf```, copy both ```kde``` to your any local folder, after that remove it.

Than reboot (i think from here, you know what problem), you have two config file

**Comment by HilariousJon** [comment]
done, then?

**Comment by thyeun** [comment]
> done, then?

reboot and see the outcome

**Comment by HilariousJon** [comment]
yes have reboot, no significant outcome :joy:

**Comment by thyeun** [comment]
@HilariousJon i cant see what you have, if you just mentioned no significant outcome, i will just assume your system die, than i cant help you at all. You have to fix it yourself.

**Comment by HilariousJon** [comment]
hmm, it just same outcome as the video I shown before . Nothing change.
> https://vimeo.com/manage/videos/1001468638

**Comment by kRHYME7** [comment]
Try to cat this it ahould contain the older sddm config

kde_settings.t2.bkp

**Comment by HilariousJon** [comment]
```
❯ cat ~/Desktop/sddm_tmp/kde_settings.t2.bkp
[Autologin]
Relogin=false
Session=
User=

[General]
HaltCommand=/usr/bin/systemctl poweroff
RebootCommand=/usr/bin/systemctl reboot

[Theme]
Current=Corners

[Users]
MaximumUid=60513
MinimumUid=1000
```

--- 

has it overwritten by this command? 
`Hyde sddm install`

**Comment by kRHYME7** [comment]
> has it overwritten by this command?
Hyde sddm install


nope, by ./install.sh initially. that's why it is good to use this dotfiles in a minimal arch install

**Comment by thyeun** [comment]
@kRHYME7 it is same file as ```00_manjaro_settings.conf``` that why i asked him keep it.

@HilariousJon check here see anything extra file here ```/usr/lib/sddm/sddm.conf.d/```

**Comment by HilariousJon** [comment]
it attains a `default.conf` here

---
> it is same file as 00_manjaro_settings.conf that why i asked him keep it

So what we are doing is to find back the config of kde sddm before, but still keep the possibility that I can reload sddm to enter into hyprland in the future? I think I may look into the electron problem in the following days. 

A lot thanks to you guys :-D

**Reply by HilariousJon** [reply]
the content of this file
```
❯ cat /usr/lib/sddm/sddm.conf.d/default.conf
[Autologin]
# Whether sddm should automatically log back into sessions when they exit
Relogin=false

# Name of session file for autologin session (if empty try last logged in)
Session=

# Username for autologin session
User=


[General]
# Which display server should be used.
# Valid values are: x11, x11-user, wayland. Wayland support is experimental
DisplayServer=x11

# Comma-separated list of environment variables to be set
GreeterEnvironment=

# Halt command
HaltCommand=/usr/bin/systemctl poweroff

# Input method module
InputMethod=

# Comma-separated list of Linux namespaces for user session to enter
Namespaces=

# Initial NumLock state. Can be on, off or none.
# If property is set to none, numlock won't be changed
# NOTE: Currently ignored if autologin is enabled.
Numlock=none

# Reboot command
RebootCommand=/usr/bin/systemctl reboot


[Theme]
# Current theme name
Current=

# Cursor size used in the greeter
CursorSize=

# Cursor theme used in the greeter
CursorTheme=

# Number of users to use as threshold
# above which avatars are disabled
# unless explicitly enabled with EnableAvatars
DisableAvatarsThreshold=7

# Enable display of custom user avatars
EnableAvatars=true

# Global directory for user avatars
# The files should be named <username>.face.icon
FacesDir=/usr/share/sddm/faces

# Font used in the greeter
Font=

# Theme directory path
ThemeDir=/usr/share/sddm/themes


[Users]
# Default $PATH for logged in users
DefaultPath=/usr/local/sbin:/usr/local/bin:/usr/bin

# Comma-separated list of shells.
# Users with these shells as their default won't be listed
HideShells=

# Comma-separated list of users that should not be listed
HideUsers=

# Maximum user id for displayed users
MaximumUid=60513

# Minimum user id for displayed users
MinimumUid=1000

# Remember the session of the last successfully logged in user
RememberLastSession=true

# Remember the last successfully logged in user
RememberLastUser=true

# When logging in as the same user twice, restore the original session, rather than create a new one
ReuseSession=true


[Wayland]
# Path of the Wayland compositor to execute when starting the greeter
CompositorCommand=weston --shell=kiosk

# Enable Qt's automatic high-DPI scaling
EnableHiDPI=true

# Path to a script to execute when starting the desktop session
SessionCommand=/usr/share/sddm/scripts/wayland-session

# Comma-separated list of directories containing available Wayland sessions
SessionDir=/usr/local/share/wayland-sessions,/usr/share/wayland-sessions

# Path to the user session log file
SessionLogFile=.local/share/sddm/wayland-session.log


[X11]
# Path to a script to execute when starting the display server
DisplayCommand=/usr/share/sddm/scripts/Xsetup

# Path to a script to execute when stopping the display server
DisplayStopCommand=/usr/share/sddm/scripts/Xstop

# Enable Qt's automatic high-DPI scaling
EnableHiDPI=true

# Arguments passed to the X server invocation
ServerArguments=-nolisten tcp

# Path to X server binary
ServerPath=/usr/bin/X

# Path to a script to execute when starting the desktop session
SessionCommand=/usr/share/sddm/scripts/Xsession

# Comma-separated list of directories containing available X sessions
SessionDir=/usr/local/share/xsessions,/usr/share/xsessions

# Path to the user session log file
SessionLogFile=.local/share/sddm/xorg-session.log

# Path to Xephyr binary
XephyrPath=/usr/bin/Xephyr

```

**Reply by kRHYME7** [reply]
@thyeun  I just moved it here so we can utilize replies.

**Comment by kRHYME7** [comment]
hey we got a discord server also a discussion tab here. Let's move this so we won't clutter the issue tab

**Comment by thyeun** [comment]
@HilariousJon Ok, now re-do everything, ```yay -S sddm``` (if show conflict with sddm-git, press Yes to remove it) your KDE is suitable using ```sddm``` repos instead of ```sddm-git``` (this is for wayland use, since your KDE is x11).

**Reply by HilariousJon** [reply]
done, and reboot, no outcome change as before.

**Reply by thyeun** [reply]
@HilariousJon Sorry, it is out of my knowledge, you have to figure out which folder or which file is control your ```sddm```. What i can figure out is the standard that been use under ```sddm```.

**Reply by HilariousJon** [reply]
@thyeun @kRHYME7 
OK, thanks for your patience. I will go two threads to find solution: one to go back to kde, other to start electron properly. I will report  the solution as long as I figure them out :-D
Again, a lot thanks to your patient reply. I am going to have a new computer, which definitely will have hyprdots on it. So I am sure that I may open another issue in this repo again. :-D

**Reply by thyeun** [reply]
@HilariousJon My two cents for you, if you getting new laptop/pc, use a distro install No-Desktop and install this Hyprdots, i'm sure you will love it. I'm enjoy it with extra enhancement (add-on waybar module, code custom css, librewolf custom css and etc). And sorry again for not possible helping you up.

**Reply by HilariousJon** [reply]
@thyeun I am in favour of endeavouros, but my new computer have the newest amd gpu, so maybe I should wait for a few months to  wait until the kernel is compatible with the hardware. One thing about arch linux, is that I am worrying about it is not that stable as manjaro :joy: But I am not that trust manjaro now :joy:

---

I have a lot to learn yet, its my fault to make everything so messy. Next time I should do more scratch first and provide information more clearly before seeking for help. So really thanks for you guys' paticience! :-)

**Reply by thyeun** [reply]
I'm Endeavouros user too.

**Reply by HilariousJon** [reply]
already noticed that :-D, in the pictures you posted before

**Comment by HilariousJon** [comment]
@thyeun @kRHYME7 
guys, for both the `vscode` and `obsidian` problem, I tried the flag:
```
--disable-accelerated-video-decode
```
This flag makes `vscode` fine. However, but it takes for about half a minute for `obsidian` to start. any idea of how to deal with this?

---

So it is like indeed the problem with my hybrid gpu. I checked my gpu in bios, and just find no way to seprarate intel and nvidia gpu....

--- 

Good news is that it also works for `chromium` and `spotify`, so I guess this can solve almost all the electron related apps problem.
Bad news is that both `spotify` and `obsidian` take about almost half  to one minute to start, which is really bad.

**Reply by HilariousJon** [reply]
output logs of `spotify`:

```
❯ spotify --disable-accelerated-video-decode
User flags:  --ozone-platform-hint=wayland
[21471:21471:0823/131449.764482:ERROR:policy_logger.cc(157)] :components/enterprise/browser/controller/chrome_browser_cloud_management_controller.cc(161) Cloud management controller initialization aborted as CBCM is not enabled. Please use the `--enable-chrome-browser-cloud-management` command line flag to enable it if you are not using the official Google Chrome build.
[21525:21551:0823/131504.811761:ERROR:ssl_client_socket_impl.cc(970)] handshake failed; returned -1, SSL error code 1, net_error -101
[21471:21471:0823/131521.614818:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=512
[21471:21471:0823/131552.632175:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=512
[21471:21471:0823/131623.655877:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=512
[21603:21610:0823/131623.836797:ERROR:command_buffer_proxy_impl.cc(131)] ContextResult::kTransientFailure: Failed to send GpuControl.CreateCommandBuffer.

```

**Reply by HilariousJon** [reply]
output logs of `obsidian`: 

```
❯ obsidian --disable-accelerated-video-decode
2024-08-23 05:06:11 Loading main app package /usr/lib/obsidian/obsidian.asar
2024-08-23 05:06:12 Checking for update using Github
2024-08-23 05:06:12 Failed to check for update using Github (net::ERR_CONNECTION_REFUSED)
Error: net::ERR_CONNECTION_REFUSED
    at SimpleURLLoaderWrapper.<anonymous> (node:electron/js2c/browser_init:2:114482)
    at SimpleURLLoaderWrapper.emit (node:events:519:28)
2024-08-23 05:06:12 Checking for update using obsidian.md
2024-08-23 05:06:13 Failed to check for update using obsidian.md (Unexpected end of JSON input)
SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at runUpdate (/usr/lib/obsidian/app.asar/main.js:183:21)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async update (/usr/lib/obsidian/app.asar/main.js:149:3)
[12421:0823/130643.086129:ERROR:command_buffer_proxy_impl.cc(131)] ContextResult::kTransientFailure: Failed to send GpuControl.CreateCommandBuffer.
[12323:0823/130643.086965:ERROR:gpu_process_host.cc(997)] GPU process exited unexpectedly: exit_code=512
[12421:0823/130713.382597:ERROR:command_buffer_proxy_impl.cc(131)] ContextResult::kTransientFailure: Failed to send GpuControl.CreateCommandBuffer.
[12323:0823/130714.194566:ERROR:gpu_process_host.cc(997)] GPU process exited unexpectedly: exit_code=512
[12323:0823/130745.300655:ERROR:gpu_process_host.cc(997)] GPU process exited unexpectedly: exit_code=512
[12421:0823/130745.553159:ERROR:command_buffer_proxy_impl.cc(131)] ContextResult::kTransientFailure: Failed to send GpuControl.CreateCommandBuffer.
```

--- 

`obsidian` is more urgent to me, its a daily must. As for `vscode` and `spotify` and `chromium`, I actually have `neovim`, online music station, and `mozilla firefox` as substitution. :joy: 

---

clicking on buttons in `spotify` may cause flicker, `obsidian` and `vscode` will not, thats good for me.

**Reply by HilariousJon** [reply]
and I guess, if I want to make `obsidian` transparent I may need to modify some css. But the slow start us more urgent to solve.

**Reply by HilariousJon** [reply]
Besides that, I also checked for the `crash log viewers`, it shows that `vscode` still crashed, but I still can start it anyway :/ weird.

**Reply by HilariousJon** [reply]
And I actually notice that, **if, I open `obsidian` in one window, I switch to another and type `obisidian` in the terminal again, it start very quick, but with the default vault starter.**

**Reply by HilariousJon** [reply]
OK, guys, so I guess this sub-problem can be marked as solved, I can run obsidian quickly with flags:

```
--disable-accelerated-video-decode
--disable-gpu
```

**Reply by kRHYME7** [reply]
Damn 😭 What's going on with that install lol. I can only see you got issues with your gpu after installing this dotfiles? Also I'm wondering if you gou supports the latest nvidia drivers because I saw you got mixed gpu drivers including Nouveau which causes conflict with dkms

maybe removing the `nvidia_drm.modeset=1` in your kernel parameter and rebooting can fix it. Or to be sure (if manjaro did not change it) find the file with ` t2.bkp `  and try to use the contents of that file... I can't tell you how to do it in grub so this is general advice.

**Reply by HilariousJon** [reply]
problem solved, I reconfigure my system proxy.

**Reply by HilariousJon** [reply]
@kRHYME7 

> Damn 😭 What's going on with that install lol. I can only see you got issues with your gpu after installing this dotfiles? Also I'm wondering if you gou supports the latest nvidia drivers because I saw you got mixed gpu drivers including Nouveau which causes conflict with dkms

I checked in bios, my gpu is mixed with intel, don't know how to seperate them. here is my `nvidia-smi`:

```

❯ nvidia-smi
Fri Aug 23 16:42:43 2024       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.107.02             Driver Version: 550.107.02     CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 4060 ...    Off |   00000000:01:00.0 Off |                  N/A |
| N/A   38C    P8              1W /   55W |      13MiB /   8188MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A       780      G   /usr/lib/Xorg                                   5MiB |
|    0   N/A  N/A       866      G   Hyprland                                        2MiB |
+-----------------------------------------------------------------------------------------+
```
currently, I am reading through some forums and docs to find way to solve `sddm` problem. I also checked back the installation process, and found nothing special :joy: maybe some logs are lost... But currently, adding `--disable-accelerated-video-decode` indeed a solution, although don't know why they still can be seen in the list of my crashed logs.

--- 

remove `nvidia_drm=1` will boot into black screen, but when I face black screen and type my password, it can directly loggin, the difference between have or not have `nvidia_drm=1` is that whether I type my password when facing a stuck asus logo or a stuck black screen, so its just like only `sddm` not showing. if I add `nomodeset` to the grub param, it will just crash.

**Comment by thyeun** [comment]
You have hybrid GPU. Same as my laptop with Nvidia + Intel. 

Try ask at manjaro forum about ```sddm```, they might have a ```.conf``` some where to let you change/configure it to get back KDE x11.

Your bios can let you choose nvidia/intel?

**Reply by HilariousJon** [reply]
No, not able it seems.

--- 

I ll go and have a post.

**Reply by HilariousJon** [reply]
@thyeun 

Actually find a way to have my kde back, though a bit funny, thing is, I can run 

```
startplasma-wayland
```
and I can get kde back in a window of hyprland. kind of like virtual machine :joy:, but blurred version 

---

Oh, but it is a kind of blur version of kde. Besides, if I run `startplasma-x11`, things will crash.

**Comment by HilariousJon** [comment]
@thyeun @kRHYME7 found the solution, we need the computer under the external graphic card mode to switch back to KDE

**Reply by HilariousJon** [reply]
thanks for all of you guys previous help. I ve been using hyprdots for more than 1 month, and quite satisfy :)
